mobile/src/main/java/be/mygod/vpnhotspot/TetheringService.kt

```diff
@@ -116,6 +124,12 @@ class TetheringService : IpNeighbourMonitoringService(), TetheringManager.Tether
         launch(Dispatchers.Main) { binder.routingsChanged() }
     }
 
+    override fun onCreate() {
+        super.onCreate()
+
+        registerReceiver(receiver, IntentFilter(TetheringManager.ACTION_TETHER_STATE_CHANGED))
+    }
```

Comment:
What's the point of this? There is no guarantee that this service will be active at any moment.

Reply:
The tethering service should be running whenever tethering is active, right? It would probably make sense to also manually call `updateTetheringState(listOf())` inside `onDestroy()`.

Reply:
No. This service is running when VPN tethering is active. Why do we need this?

Reply:
Sorry, that's what I meant. It's so that Tasker can receive events from VPN Hotspot whenever its tethered interfaces change (e.g., you could set up a Tasker profile to re-enable USB tethering in VPN Hotspot if it gets disabled automatically for whatever reason).

Reply:
Okay, then it wouldn't work properly unless VPN tethering happens to be running?

Reply:
Right, but that's the point of VPN Hotspot. The events aren't for general tethering state; those are already built into Tasker.

Reply:
Okay just to clarify, is this thing supposed to report the list of tethered interfaces or the list of *VPN* tethered interfaces?

Reply:
It should be VPN tethered interfaces. I don't know for sure if I'm using the right value.

Reply:
Got it. I'll fix it for you then.